src/tests/testRefCount.cc

```diff
@@ -55,8 +55,8 @@ testRefCount::testCountability()
         ToRefCount anObject(new _ToRefCount);
         CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
         CPPUNIT_ASSERT_EQUAL(1, anObject->someMethod());
-        anObject = anObject;
-        CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
+        // anObject = anObject;  //-Wself-assign-overloaded will prevents this
+        // CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
```

Comment:
If possible, please use a different workaround suggested in clang [documentation](https://releases.llvm.org/7.0.0/tools/clang/docs/ReleaseNotes.html#improvements-to-clang-s-diagnostics): 

> the warning can be suppressed by adding *& to the right-hand side

If that works, I would still add some kind of "Casting is to prevent -Wself-assign-overloaded warnings" source code comment, of course.

Reply:
done